server/pulp/server/webservices/views/consumers.py

```diff
@@ -0,0 +1,1047 @@
+from pulp.server.webservices.controllers.decorators import auth_required
+from pulp.server.webservices import serialization
+from pulp.server.webservices.views.util import (_ensure_input_encoding,
+                                                generate_json_response,
+                                                generate_json_response_with_pulp_encoder,
+                                                generate_redirect_response,
+                                                json_body_required,
+                                                json_body_allow_empty)
+
+
+def expand_consumers(options, consumers):
+    """
+    Expand a list of users based on the flag specified in the query parameters.
+    The _href is always added by the serialization function used.
+    Supported options:
+      details = include details
+      bindings - include bindings
+
+    :param options: The (expanding) options.
+    :type options: dict
+    :param consumers: A list of consumers
+    :type consumers: list
+
+    :return: A list of expanded consumers.
+    :rtype: list
+    """
+    details = (options.get('details', 'false')).lower()
+    bindings = (options.get('bindings', 'false')).lower()
+    if details != 'false':
+        bindings = 'true'
+    # add bindings
+    if bindings != 'false':
+        ids = [c['id'] for c in consumers]
+        manager = factory.consumer_bind_manager()
+        criteria = Criteria({'consumer_id': {'$in': ids}})
+        bindings = manager.find_by_criteria(criteria)
+        collated = {}
+        for b in bindings:
+            lst = collated.setdefault(b['consumer_id'], [])
+            lst.append(b)
+        for _consumer in consumers:
+            _consumer['bindings'] = \
+                [serialization.binding.serialize(b, False)
+                    for b in collated.get(_consumer['id'], [])]
+    return consumers
+
+
+class ConsumersView(View):
+    """
+    View for consumers.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request):
+        """
+        List the available consumers.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+
+        :return: Response containing a list of consumers
+        :rtype: django.http.HttpResponse
+        """
+        query_params = request.GET
+
+        manager = factory.consumer_query_manager()
+        consumers = expand_consumers(query_params, manager.find_all())
+        for consumer in consumers:
+            add_link(consumer)
+        return generate_json_response_with_pulp_encoder(consumers)
+
+    @auth_required(authorization.CREATE)
+    @json_body_required
+    def post(self, request):
+        """
+        Create a consumer and return a serialized object containing just created consumer.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+
+        :raises: MissingValue if ID is not provided
```

Comment:
I was recently informed that according to sphinx docs, the second colon should be after the exception:

    :raises MissingValue: if ID is not provided